src/canister/individual_user_template/src/api/device_id_management/add_device_id.rs

```diff
@@ -0,0 +1,39 @@
+use ic_cdk::api;
+use ic_cdk_macros::update;
+use shared_utils::canister_specific::individual_user_template::types::device_id::DeviceIdentity;
+
+use crate::{
+    data_model::CanisterData,
+    CANISTER_DATA
+};
+
+/// #### Access Control
+/// Only the user whose profile details are stored in this canister can add the device identity.
+#[update]
+fn add_device_id(identity_token: String) -> Result<bool, ()> {
+    // * access control
+    let current_caller = ic_cdk::caller();
```

Comment:
Check this on how to do access control - https://github.com/go-bazzinga/hot-or-not-backend-canister/blob/7fb79beac92a49343bffd778882d681cf0a523d5/src/canister/individual_user_template/src/api/post/add_post_v2.rs#L25